src/app/modules/item/http-services/create-item.service.ts

```diff
@@ -10,13 +10,11 @@ interface NewItemData {
   id: string;
 }
 
-export interface NewItem {
+export type NewItem = {
   title: string,
   type: ItemType,
-  language_tag: string,
-  parent?: { item_id: string },
-  as_root_of_group_id?: string
-}
+  language_tag: string
+} & ({ parent: {item_id: string} } | { as_root_of_group_id: string })
```

Comment:
Good but this way you cannot have both `parent` and `as_root_of_group_id`, which is possible in the service.

Reply:
I haven't seen issues when I tested it, and according to [exemple](https://www.typescriptlang.org/play/?ssl=32&ssc=87&pln=20&pc=1#code/PTCwCgQAgSQOQMIBkCqBlGA1AolA8gEoQgQQAuAngA4CmUAsjQHYCuMZNAtngE5QC8UAN4QoYqGQCWZADY0AXFADOZHpKYBzANyjxkgMYB7JopVrNO8AF8oAMigAKIVCOcqx5mUWGARgCsafTIoGwAfYRcZAwBrRR9DQzkAQyYQgEooaB8eJP1omjIlKCSeOkk3Qx4yFK87KAALJKKqUv0aABNmNqhDADcaPlDScDlgjhVeRUZWdi5eS3GyXgEIqVkFKAAiMk2AGigDY0VNyU2QzOAoAZ5K-aYPYtLlAvIaCb5BZzW5Y539w5MW1O%201c7iYnkUQhsNmggBRyV7vFZfaQ-LZ-A5GQEnPaRGKKVQsOgwy7w8CLZafCQojbbHEA47A3F5fE8QkgwwVcFMWpQ85w4YkSCXbAADWQ6CwuEIxDA4HI1Do0zYHE4IsqSN0Ym%20GzM6m0moxR2Uqj1lhs9gcBvEUHCzlBHm53n8gTIWiZ0QA-Ipwf0%20FYreJbS4OWDPF6oD6Bm79FFmVB4okaCkQga0sNRhI3mQ1TwpsxlVwcwsszmkVT1r86ZiGWdiVceDceHcHiU6EoXmSS%20rKdrK-9q0CcfauTzoRcoKTFqWe9S%204asYyY3iJKyiePJ12PqtZ2iq0bsSDY7EV2zg5yIcIx9Brrd44YyPVHm2CkA) found online, it should be working